backend/src/main/kotlin/clarity/backend/entity/CardSetEntity.kt

```diff
@@ -318,4 +325,81 @@ class CardSetEntity() {
             )
         }
     }
+
+    fun getCardSetsOrderedByLikes() : GetCardSetsOrderedByLikesResponse {
+        val db = DataManager.conn()
+        try {
+            val statement = db!!.createStatement()
+            val query = "SELECT * FROM CardSet ORDER BY likes DESC;"
```

Comment:
Should we only be checking for public CardSets here?

Reply:
Ah right, I guess that makes sense. But then why do we have the likes attribute as NOT NULL? Is there any relevance to likes for private sets?

Reply:
Yeah I suppose it wouldn't, though maybe the logic of changing it would be easier if it was set to 0 instead of null for private sets, since we don't have to worry about null checking anymore, and just +1/-1 the counter

Reply:
> Ah right, I guess that makes sense. But then why do we have the likes attribute as NOT NULL? Is there any relevance to likes for private sets?

A set can go from public to private and vice versa. It could be public one day, get a couple of likes, and then go private (and if it went back public it should retain those likes)

Reply:
Ok, makes sense. I'll make this change.

Reply:
Done.